lib/ops/opsservice/install.go

```diff
@@ -128,16 +128,19 @@ func (s *site) createInstallExpandOperation(operationType, operationInitialState
 	}
 
 	tokenType := storage.ProvisioningTokenTypeInstall
+	expires := s.clock().UtcNow().Add(defaults.InstallTokenTTL)
 	if op.Type == ops.OperationExpand {
 		tokenType = storage.ProvisioningTokenTypeExpand
+		// Do not expire join tokens
+		expires = time.Time{}
```

Comment:
Won't they keep accumulating for every join operation then? As an easy way to clean them up, we could loop over all expand tokens before creating an expand operation and delete the ones which are not for an in-progress expands.

Reply:
Do we want to clean them up? AFAIC, there's no reliable way to tell whether the operation is complete if it's failed, hence we don't remove the operation but merely mark it as failed - why remove the token, it is as good a history as the operation itself?